### PR TITLE
[BC] Update `explore_at_state_generator` with new exploration strategy

### PR DESCRIPTION
**Pull request by tvmarino**
Updating the exploration strategy as discussed with @alekh.
The strategy works as follows. At the current exploration state,
the exploration policy will return an array of logits. The logit of
the action currently taken is set to `-np.Inf` and an exploration action
is sampled from the remaining available actions. This update required
updating the tests for `ModuleExplorerWorker`. Additionally, added
`module_explorer_type` to `ModuleWorker` to enable overriding the
exploration strategy by overriding `explore_at_state_generator` of
`ModuleExplorer` class.

### CODE REVIEW

**Review by alekh — APPROVED** [review]
Looks to be in line with the proposed exploration scheme. Thanks!
